src/QueryType/Update/RequestBuilder.php

```diff
@@ -270,7 +270,13 @@ private function buildFieldsXml(string $key, ?float $boost, $value, ?string $mod
         if (\is_array($value)) {
             $nestedXml = '';
             foreach ($value as $multival) {
-                if (\is_array($multival)) {
+                if (\is_array($multival) && $key === '_childDocuments_') {
```

Comment:
```suggestion
                if (\is_array($multival) && '_childDocuments_' === $key) {
```

Solarium follows the Symfony Coding Standards and uses [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions).